src/index.ts

```diff
@@ -13,8 +13,13 @@ import createImpl, {
 export * from './vanilla'
 
 // For server-side rendering: https://github.com/react-spring/zustand/pull/34
-const useIsoLayoutEffect =
-  typeof window === 'undefined' ? useEffect : useLayoutEffect
+// Deno support: https://github.com/pmndrs/zustand/issues/347
```

Comment:
```suggestion
// Deno support: https://github.com/pmndrs/zustand/issues/347
// In the case of Deno, we'll check the globalThis for the existance of the Deno object to see if we are server side
const isGlobalThisDeno = globalThis && globalThis.Deno ? true : false;
```

Reply:
> `globalThis &&`

I assume you're doing this just to be sure `globalThis` is defined? But in case `globalThis` doesn't exist (don't even know if it's possible), this code will throw.

Something like this should be safer:
```js
const isGlobalThisDeno = typeof globalThis !== "undefined" && !!globalThis.Deno;
```

Reply:
tbh, I don't want to check both `window` and `globalThis`. Let me add a commit with the current approach, which should have a small impact on bundle size.